### ISSUE TITLE
Sagemaker badly interpret and plot training metrics with very small numerical values.

### ISSUE THREAD

**Issue by adekunleoajayi** [aws/sagemaker-python-sdk#4263]
**Describe the bug**
I am currently fine-tuning llama-2 models using HuggingFace transform with Sagemaker training estimator. To monitor training performance, I defined relevant metrics using the metric definition. As expected the training pipeline was executed and ran successfully but some of the tracked metrics values were badly interpreted by Sagemaker for cases where the values are very small.

For example, if the numerical value is 7E-5, the exponent is dropped and only the coefficient is plotted and transmitted via training job API making both the plotted and transmitted values wrong.

To validate that this is indeed a bug, I decided to save the training history to disk, and then compare it with the automatically tracked metrics by Sagemaker.

**Screenshots or logs**

_**1 - A comparison plot for validation loss generated by Sagemaker TrainingJobAnalytics vs saved from training history**_
<img width="560" alt="Screenshot 2023-11-20 at 16 36 38" src="https://github.com/aws/sagemaker-python-sdk/assets/12529807/53bd746b-5329-4d87-bb91-10ca922eecd6">

_**2 - A comparison of the actual numerical values for validation loss generated by Sagemaker TrainingJobAnalytics vs saved from training history**_
<img width="377" alt="Screenshot 2023-11-20 at 16 36 28" src="https://github.com/aws/sagemaker-python-sdk/assets/12529807/cf8503ae-691c-41b6-8b0d-6b16144cd73e">

_**3 - CloudWatch logs for this particular training job**_
<img width="1278" alt="Screenshot 2023-11-21 at 10 29 43" src="https://github.com/aws/sagemaker-python-sdk/assets/12529807/75c03084-7b06-40f7-b197-db3ce686bb8d">

_**4 - A plot of the validation loss automatically generated by Sagemaker Training Job monitor.**_
<img width="397" alt="Screenshot 2023-11-20 at 16 38 50" src="https://github.com/aws/sagemaker-python-sdk/assets/12529807/dc1838a2-9c80-4378-97d1-9e6f7aeed4d4">

**System information**
A description of your system. Please provide:
- **SageMaker Python SDK version**:
- **Framework name (eg. PyTorch) or algorithm (eg. KMeans)**:
- **Framework version**:
- **Python version**:
- **CPU or GPU**:
- **Custom Docker image (Y/N)**:

**Additional context**

**Comment by martinRenou** [comment]
Thank you for opening an issue. Would you be able to provide a small example that replicates this issue?

**Comment by martinRenou** [comment]
I'm closing this issue for triaging. Feel free to continue the discussion and reopen if you're able to provide a code snippet for reproduction.

Thank you!
